### DISCUSSION TITLE
RTL Support - Core

### DISCUSSION THREAD

**Discussion by siddharth1903** [vmware/clarity#6272] [Q&A] [answered]
Hey Guys,
Wanted to understand if BiDi support is something that is planned for Clr core components. 

Regards,
Sid

**Comment by mathisscott** [comment, marked as answer]
BiDi support? I'm unfamiliar with that term.

As for RTL support, largely yes. Our [forms in Clarity Core already support RTL](https://clarity.design/storybook/core/?path=/story/stories-forms--internationalization).

As for the other components, most of the layout is handled by the end user. There are certainly some areas where we could improve RTL support and we'd appreciate any issues/bugs opened towards that end.

**Reply by siddharth1903** [reply]
@mathisscott apologies .. yes (bidi)meant RTL and LTR... Couldnt find any documentation around that... will follow the story book to understand more... Is there a list where you guys can highlight the components that are currently supported for rtl vs the ones that still need to be tested/certified.. we can certainly help with issues/bugs

**Reply by mathisscott** [reply]
Right now, I think the form controls are the extent of our RTL support. It emerged as an experiment to validate whether our users wanted RTL support and how we could do it.

Internally, we've had debates about what could/should support RTL (in terms of components). But, frankly, we are not experts in the specifics. That's where the community reporting on the state of our support would be helpful. If you find, say, the core alert isn't RTL and you would like it to be, we would welcome that request. We might need some direction related to how that could look but we are definitely willing to implement it.

**Reply by siddharth1903** [reply]
That's great @mathisscott ..we will be able to report these to you guys and looks like a strategy that could work for both the worlds.. We are at the moment thinking about migrating our own internal design systems from clr angular to core, in parts as suggested in a different thread. Once we are done, we can run a sanity test with rtl and report back...
